### PR TITLE
More Kubernetes features and additions

### PR DESCRIPTION
**Pull request by adelbertc**
Initially started as just implementing volume mounts, but added some other stuff too..

- Add support for empty "scratch" directory volume mounts
- Add support for K8s 1.9.x and 1.10.x
- Allow operators to choose between in or out of cluster - if deploying in-cluster it will assuming the appropriate RBAC is configured and use Pod-mounted certs
- Build image to run as non-root
- Stub Consul client in K8s mode so we don't get bogus Consul errors in logs

### CODE REVIEW

File: core/src/main/scala/Config.scala [hunk 1]
```diff
@@ -654,29 +638,49 @@ object Config {
...
             consulClient  <- consul
             sched         <- readNomadScheduler(schedConfig.subconfig("nomad"))
             healthChecker = health.Http4sConsulHealthClient(consulClient)
-          } yield (sched, healthChecker)
+          } yield (sched, healthChecker, consulClient)
+
         case Some("kubernetes") =>
           readKubernetesClient(schedConfig.subconfig("kubernetes")).map { client =>
-            (new scheduler.KubernetesHttp(client), health.KubernetesHealthClient(client))
+            (new scheduler.KubernetesHttp(client), health.KubernetesHealthClient(client), StubbedConsulClient)
```
Comment (timperrett) [inline core/src/main/scala/Config.scala]: `StubbedConsulClient`? Would be good to understand the rationale here.

Comment (adelbertc) [reply core/src/main/scala/Config.scala]: I confess this was selfishly motivated - we are not using Consul with Kubernetes not are we (or you afaik) planning on it, but in retrospect it is totally possible (I think [Stripe](https://stripe.com/blog/operating-kubernetes) maybe does this?).

Anyways assuming that is the case, there is no sane values to put for the Consul portion of the config (which we need to make more generic BTW) and so when Nelson goes to refresh routing tables you get this nice stack trace in the logs that is bogus.

Unsure what to do here:

- The Correct Way™ would be to give an option in the config to signal what backend to use for the routing layer, similar to what we do for the scheduler
- Try to make a dummy call to Consul and if that fails fallback to the stubbed client

This of course is a stopgap while we're in the "Kubernetes without Consul currently has no routing support from Nelson" stage. Curious to hear your thoughts

File: core/src/test/resources/nelson/manifest.v1.everything.yml [hunk 2]
```diff
@@ -108,6 +108,11 @@ plans:
       - ref: s3
         uri: http://s3.aws.com
     expiration_policy: retain-latest-two-major
+    volumes:
+      - name: an-empty-dir
+        mountPath: /foo/bar
+        emptyDir:
```
Comment (timperrett) [inline core/src/test/resources/nelson/manifest.v1.everything.yml]: I'm not sure we should be specific about the kind of volume here based on our discussion at lunch. If we go for templating later, then we can simply have the user specify the size of the volume they'd like. WDYT?

**Comment by timperrett** [comment]
Other than that one comment, I think this looks fine 😄

**Comment by adelbertc** [comment]
@timperrett Made the change, OK to merge?

**Pull request merged**